vscode/src/telemetry.ts

```diff
@@ -0,0 +1,56 @@
+import * as vscode from "vscode";
+import TelemetryReporter from "@vscode/extension-telemetry";
+import { log } from "qsharp";
+// the application insights key (also known as instrumentation key)
+const key = "175861b7-3a41-4015-9571-1d930b8b0722";
+
+// telemetry reporter
+let reporter: TelemetryReporter | undefined;
+
+export enum EventType {
+  DebugSessionStart = "DebugSessionStart",
+  Initialize = "Initialize",
+  LoadLanguageService = "LoadLanguageService",
+}
+
+type Empty = { [K in any]: never };
+
+type EventTypes = {
+  [EventType.Initialize]: {
+    properties: Empty;
+    measurements: Empty;
+  };
+  [EventType.DebugSessionStart]: {
+    properties: Empty;
+    measurements: {
+      timeToStart: number;
+    };
+  };
+  [EventType.LoadLanguageService]: {
+    properties: Empty;
+    measurements: {
+      timeToStart: number;
```

Comment:
In general when dealing with durations I like putting the units in the property name e.g. `timeToStartMs`